DebianRepoChecker: improve flexibility of Debian repo checking

ramcq opened the pull request:
Add support for .bz2 Packages files, and repos with an exact URL which are usually specified in sources.list with just a base plus the subdirectory where Packages.* can be found.

ramcq committed:
DebianRepoChecker: add support for Packages.gz2

Also, prefer .xz over .bz2 over uncompressed.

ramcq committed:
DebianRepoChecker: add support for exact repo URLs

Debian upstream repos have the dists/{dist}/../{component} structure, but apt
sources.list also supports a syntax with a base URL followed by a subdirectory:

  "suite can specify an exact path, in which case the components must be
  omitted and suite must end with a slash (/). This is useful for the case when
  only a particular sub-directory of the archive denoted by the URI is of
  interest. If suite does not specify an exact path, at least one component
  must be present."

Add support for omitting the "component" if "dist" ends in / to allow checking
repos with this structure.

joaquimrocha commented on src/checkers/debianrepochecker.py:
```diff
@@ -155,8 +156,12 @@ def _split_packages_from_page(self, packages_page):
     def _get_package_from_url(self, package_name, root, dist, component, arch):
-        packages_url = DEB_PACKAGES_URL.format(root=root, dist=dist,
-                                               comp=component, arch=arch)
+        if dist[-1:] == '/' and component is None:
```
You shouldn't need the `:` after the `-1` here. A more pythonic way for this could be also `dist.endswith('/')` .

joaquimrocha replied:
Also, should we fail when the dist has not slash at the end but the component is still `None`? I am just guessing it is pretty easy to forget the slash at the end, in which case we'd be formatting `packages_url` with a `None` component.

ramcq replied:
Sorted.

ramcq committed:
DebianRepoChecker: print a warning if component is missing

Now that component is optional, if dist does not end in a / specifying
an exact path, then the configuration is invalid. Detect this case and
print a warning.
